internal/plugin.go

```diff
@@ -101,6 +109,113 @@ func (p *Plugin) Config(config []byte, baseDirectory string) error {
 	return p.assertValidConfig()
 }
 
+// addFieldNotFoundHelp adds help text to errors that occur when the input config
+// has a field that is not present in the Config struct (eg a typo), to assist
+// the user in debugging the problem. If the input error was not from a missing
+// field, then it is returned unchanged.
+func addFieldNotFoundHelp(err error) error {
```

Comment:
Could you please add these functions to a new file so that `plugin.go` doesn't keep growing?